app/models/extract_fetcher.rb

```diff
@@ -36,62 +29,55 @@ def fetch_minimal?
     @strategy == :fetch_minimal
   end
 
-  def fetch_users?
-    @topic == :reduce_by_user
-  end
-
-  def fetch_subjects?
-    @topic == :reduce_by_subject
-  end
-
   def user_extracts
-    return Extract.none if filter.fetch(:user_id, nil).blank?
+    return Extract.none if query.fetch(:user_id, nil).blank?
 
-    corrected_filter = filter.except(:subject_id)
+    corrected_query = query.except(:subject_id)
     if fetch_minimal?
-      Extract.where(corrected_filter.merge(id: @extract_ids))
+      Extract.where(corrected_query.merge(id: @extract_ids))
     else
-      Extract.where(corrected_filter)
+      Extract.where(corrected_query)
     end
   end
 
   def subject_extracts
-    corrected_filter = filter.except(:user_id)
+    corrected_query = query.except(:user_id)
 
     exact_subject_ids = Extract
       .find(@extract_ids)
       .pluck(:subject_id)
-      .append(filter[:subject_id])
+      .append(query[:subject_id])
       .uniq
 
     augmented_subject_ids = augment_subject_ids(exact_subject_ids)
 
     if fetch_minimal?
-      get_minimal_subject_extracts(corrected_filter, augmented_subject_ids, exact_subject_ids)
+      get_minimal_subject_extracts(corrected_query, augmented_subject_ids, exact_subject_ids)
     else
-      get_all_subject_extracts(corrected_filter, augmented_subject_ids)
+      get_all_subject_extracts(corrected_query, augmented_subject_ids)
     end
   end
 
-  def get_minimal_subject_extracts(filter, augmented_subject_ids, exact_subject_ids)
+  def get_minimal_subject_extracts(query, augmented_subject_ids, exact_subject_ids)
     # is an extract exactly in the list of extracts with the specified subject
     # or is an extract for one of those prior subjects but not the specified subjects
-    Extract.where(filter.merge(id: @extract_ids))
+    Extract.where(query.merge(id: @extract_ids))
       .or(Extract.where(
-        filter
+        query
           .except(:subject_id)
           .merge(subject_id: augmented_subject_ids-exact_subject_ids)
       )
     )
   end
 
-  def get_all_subject_extracts(filter, augmented_subject_ids)
+  def get_all_subject_extracts(query, augmented_subject_ids)
     # is an extract for any of the subjects that were mentioned or any of their parents
-    Extract.where(filter.except(:subject_id).merge(subject_id: augmented_subject_ids))
+    Extract.where(query.except(:subject_id).merge(subject_id: augmented_subject_ids))
   end
 
   def augment_subject_ids(id_list)
     additional_linked_subject_ids = id_list.map do |subject_id|
+      next if subject_id.blank?
```

Comment:
same as above, i didn't want to touch this code because i didn't write it. i can apply the suggested fix and see how it affects the test suite but i wonder whether it's in scope for this refactor.

Reply:
Created an issue for this specific refactor: https://github.com/zooniverse/caesar/issues/967